### PR TITLE
1372 patient list items

### LINKED ISSUE #1372: [Quest/Malawi Core] Patient List items
**Name of feature to enhance**
Patient List items

related to https://github.com/opensrp/fhir-resources/issues/87

**Description**
- The patient list item should have a unique identifier (ART number, HTS number or HCC number depending on the type of patient), a title with their name, a subtitle with their age
- We need to show icons for male patients, female non-pregnant patient, female pregnant patient, female patient with exposed infant and exposed infant

Image

Eg.

Image

**Transition**

Image

Image

### PR DESCRIPTION
**Pull request by LZRS**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #1372 

**Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by LZRS** [comment]
image 2022-06-24

**Review by AbdulWahabMemon — APPROVED** [review]
lgtm

**Comment by KhumboLihonga** [comment]
Hey @LZRS, awesome work. One small adjustment, for a community positive patient, please change the subtitle label of the patient type from "Community Positive" to "No Conf Test"

File: android/quest/src/main/java/org/smartregister/fhircore/quest/util/mappers/RegisterViewDataMapper.kt [hunk 1]
```diff
@@ -89,20 +96,39 @@ class RegisterViewDataMapper @Inject constructor(@ApplicationContext val context
...
       is RegisterData.HivRegisterData ->
         RegisterViewData(
           logicalId = inputModel.logicalId,
-          title = listOf(inputModel.name, inputModel.age).joinToString(", "),
-          subtitle = inputModel.gender.translateGender(context).capitalizeFirstLetter(),
-          identifier = inputModel.identifier ?: ""
-        )
+          title = inputModel.name,
+          subtitle = "${inputModel.age}, ${inputModel.healthStatus.display}",
+          //          subtitle = inputModel.gender.translateGender(context).capitalizeFirstLetter(),
```
Comment (AbdulWahabMemon) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/util/mappers/RegisterViewDataMapper.kt:107]: comment out code can be remove

File: android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/PatientExtension.kt [hunk 2]
```diff
@@ -224,9 +224,20 @@ fun Patient.extractOfficialIdentifier(): String? =
...
+        tagList.isEmpty() ||
+        tagList[0].code == null ||
+        tagList[0].code.isEmpty()
+    )
       return HealthStatus.DEFAULT
-    HealthStatus.valueOf(tagList[0].code?.uppercase(Locale.getDefault())?.replace("-", "_") ?: "")
+    HealthStatus.valueOf(tagList[0].code!!.uppercase(Locale.getDefault()).replace("-", "_")).apply {
+      display =
+        when (this) {
+          HealthStatus.NEWLY_DIAGNOSED_CLIENT, HealthStatus.CLIENT_ALREADY_ON_ART -> "ART Client"
```
Comment (AbdulWahabMemon) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/PatientExtension.kt:236]: hard string can be use from string resource

**Review by AbdulWahabMemon — DISMISSED** [review]

**Pull request merged**